CAM-11669: use latest OpenJPA 2.x patch level

ThorbenLindhauer opened the pull request:
related to CAM-11669

koevskinikola commented on parent/pom.xml:
```diff
@@ -164,6 +164,11 @@
+      <dependency>
+        <groupId>org.hibernate.javax.persistence</groupId>
+        <artifactId>hibernate-jpa-2.0-api</artifactId>
+        <version>1.0.1.Final</version>
```
It would be keeping with the pattern if we move the version in a property defined in the `properties` section above.

ThorbenLindhauer replied:
Not sure this is really a pattern we have, it's rather 50:50 when you look at all the dependencies. If we need the version as a property, we can still change that in the future.

koevskinikola commented on parent/pom.xml:
```diff
@@ -164,6 +164,11 @@
         <version>1.0</version>
         <scope>provided</scope>
```
1. Is the `provided` scope really needed in the `dependencyManagement` section?
  * If it can be removed, then we should also remove the `test` scope on the `openjpa` dependency definition here (line 219).
2. Do we also need to update the dependency here as in the `engine/pom.xml`?

ThorbenLindhauer replied:
Re 1: It's not needed.
Re 2: This is a dependency management section. If we still use the dependency in any inherting module, we should keep it here. If we don't use it anymore we can remove it. We have case 1, we use `persistence-api` still in engine-spring where we test in the case of Spring 3 against a Hibernate version that implements JPA 1.0.

koevskinikola commented on engine/pom.xml:
```diff
@@ -103,11 +103,11 @@
-      <groupId>javax.persistence</groupId>
-      <artifactId>persistence-api</artifactId>
+      <groupId>org.hibernate.javax.persistence</groupId>
+      <artifactId>hibernate-jpa-2.0-api</artifactId>
```
Should we document this change in the migration guide, or it would not impact users at all?

ThorbenLindhauer replied:
I don't think this needs a mention. JPA 2.0 is part of Java EE 6 and Java EE 6 was e.g. already supported by JBoss 7.2.

ThorbenLindhauer commented:
@koevskinikola I removed the scopes from the dependency management section in the parent pom. For the other hints I have added comments on why I don't think we need to make changes right now.

koevskinikola commented:
Looks good! Feel free to merge.
